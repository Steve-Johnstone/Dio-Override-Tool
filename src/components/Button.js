import PropTypes from 'prop-types';
import 'bulma/css/bulma.min.css';

const Button = ({ className, name, id, text, onClick, icon, link }) => {
	return (
		<div>
			<button
				className={`button ${className}`}
				id={id}
				name={name}
				onClick={onClick}
			>
				{icon && <span id='icon-span'>{icon} &nbsp;</span>}
				{text}
				{link && link}
			</button>
		</div>
	);
};

Button.propTypes = {
	className: PropTypes.string,
	name: PropTypes.string,
	id: PropTypes.string,
	text: PropTypes.string,
	onClick: PropTypes.func,
	icon: PropTypes.object,
	link: PropTypes.object,
};

export default Button;
