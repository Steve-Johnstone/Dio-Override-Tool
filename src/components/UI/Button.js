import PropTypes from 'prop-types';
import 'bulma/css/bulma.min.css';

const Button = ({ className, name, id, text, onClick, icon }) => {
	return (
		<div>
			<button
				className={`button ${className}`}
				id={id}
				name={name}
				onClick={onClick}
			>
				{icon && <span id='icon-span'>{icon} &nbsp;</span>}
				{text}
			</button>
		</div>
	);
};

Button.propTypes = {
	className: PropTypes.string,
	name: PropTypes.string,
	id: PropTypes.string,
	text: PropTypes.string,
	onClick: PropTypes.func,
	icon: PropTypes.object,
};

export default Button;
