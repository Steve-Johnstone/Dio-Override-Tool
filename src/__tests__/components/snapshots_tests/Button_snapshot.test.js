import renderer from 'react-test-renderer';
import Button from '../../../components/UI/Button';

describe('Button component', () => {
	const onClick = jest.fn();

	it('should display the DOM correctly', () => {
		const tree = renderer
			.create(
				<Button
					className='is-info'
					name='copy-button'
					id='copy-button'
					text='Copy the URL'
					onClick={onClick}
					// icon={{}}
				/>
			)
			.toJSON();

		expect(tree).toMatchSnapshot();
	});
});
