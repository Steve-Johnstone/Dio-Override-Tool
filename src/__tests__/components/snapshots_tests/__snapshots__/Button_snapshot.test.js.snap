// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should display the DOM correctly 1`] = `
<div>
  <button
    className="button is-info"
    id="copy-button"
    name="copy-button"
    onClick={[MockFunction]}
  >
    Copy the URL
  </button>
</div>
`;
